File: android/src/main/java/fr/lelab/phonelocked/CapacitorPhoneLockedPlugin.java
package fr.lelab.phonelocked;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorPhoneLocked")
public class CapacitorPhoneLockedPlugin extends Plugin {

    private CapacitorPhoneLocked implementation = new CapacitorPhoneLocked();

    @PluginMethod
    public void checkPhoneLocked(PluginCall call) {

        JSObject ret = new JSObject();


        ret.put("value", implementation.checkPhoneLocked(getContext()));
        call.resolve(ret);
    }
}

File: ios/Sources/CapacitorPhoneLockedPlugin/CapacitorPhoneLockedPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorPhoneLockedPlugin)
public class CapacitorPhoneLockedPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CapacitorPhoneLockedPlugin"
    public let jsName = "CapacitorPhoneLocked"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "checkPhoneLocked", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = CapacitorPhoneLocked()

    @objc func checkPhoneLocked(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.checkPhoneLocked()
        ])
    }
}
